File: frontend/src/types/category.ts
export interface CategoryItem {
  id: number;
  name: string;
  description?: string | null;
  parent_id?: number | null;
  question_count: number;
  created_at: string;
  updated_at: string;
}

export interface CreateCategoryPayload {
  name: string;
  description?: string | null;
  parent_id?: number | null;
}

export interface UpdateCategoryPayload {
  name?: string;
  description?: string | null;
  parent_id?: number | null;
}

File: backend/app/models/category.py
"""
分类模型
"""
from sqlalchemy import Column, Integer, String, Text, DateTime, UniqueConstraint, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from app.core.database import Base


class Category(Base):
    """分类表"""
    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    name = Column(String(100), nullable=False)
    description = Column(Text, nullable=True)
    parent_id = Column(Integer, ForeignKey("categories.id", ondelete="SET NULL"), nullable=True, index=True)
    question_count = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # 关系
    user = relationship("User", back_populates="categories")
    questions = relationship("Question", back_populates="category")
    parent = relationship("Category", remote_side=[id], backref="children")

    __table_args__ = (
        UniqueConstraint('user_id', 'name', name='uq_user_category_name'),
    )
